metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala

```diff
@@ -901,6 +901,36 @@ class MetalsLanguageServer(
       referencesResult(params).locations.asJava
     }
 
+  def typeDefinition(
+      params: TextDocumentPositionParams,
+      token: CancelToken
+  ): Future[java.util.List[Location]] = {
+    compilers
+      .getLocation(params, token, interactiveSemanticdbs)
+      .flatMap {
+        case Some(loc) =>
+          Future(List(loc).asJava)
+        case None =>
```

Comment:
We can remove the None branch now since we do everything in the presentation compile.

Reply:
Sometimes presentation compiler failed to correctly resolve type so I use definitionOrReferences as fallback method.

Reply:
We can't use `definitionOrReferences` since that can give us references when we are at definition. So at value definition it will find all references to it and that will produce some random behaviour. 

If we cannot resolve type and we are not at the definition that can also be done in the presentation compiler.